x/incentives/keeper/distribute_test.go

```diff
@@ -2080,3 +2099,121 @@ func (s *KeeperTestSuite) setupVolumes(poolIds []uint64, updatedPoolVolumes []os
+			coinsDistributed: defaultCoins,
+		},
+		"3: non-perpetual gauge - deleted": {
+			groupGauge:       withNonPerpetualEpochs(withIsPerpetual(deepCopyGauge(defaultPerpetualGauge), false), 0, 1),
+			coinsDistributed: defaultCoins,
+		},
+
+		"4: non-perpetual gauge - deleted (non-zero community pool update)": {
+			// Note that total coins in gauge = 2x defaultCoins
+			// Already distributed coins = none
+			// Coins to distribute = defaultCoins
+			// Therefore, the community pool should receive 2x defaultCoins - defaultCoins = defaultCoins
+			groupGauge:       withCoinsToDistribute(withNonPerpetualEpochs(withIsPerpetual(deepCopyGauge(defaultPerpetualGauge), false), 0, 1), defaultCoins.Add(defaultCoins...), sdk.NewCoins()),
+			coinsDistributed: defaultCoins,
+			amountToPrefund:  defaultCoins,
+		},
+
+		// // edge cases
+
+		"5: incentives module does not have enough balance to transfer dust to community pool": {
+			// Note that total coins in gauge = 2x defaultCoins
+			// Already distributed coins = none
+			// Coins to distribute = defaultCoins
+			// Therefore, the community pool should receive 2x defaultCoins - defaultCoins = defaultCoins
+			// However, there is no balance in the incentives module, leading to error.
+			groupGauge:       withCoinsToDistribute(withNonPerpetualEpochs(withIsPerpetual(deepCopyGauge(defaultPerpetualGauge), false), 0, 1), defaultCoins.Add(defaultCoins...), sdk.NewCoins()),
+			coinsDistributed: defaultCoins,
+			amountToPrefund:  emptyCoins,
+
+			expectError: true,
+		},
+		"6: perpetual gauge with a 2x coinsDistributed value - updated": {
+			groupGauge:       defaultPerpetualGauge,
+			coinsDistributed: defaultCoins.Add(defaultCoins...),
+		},
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.SetupTest()
+			incentivesKeeper := s.App.IncentivesKeeper
+
+			// Setup group gauge to confirm that it is deleted if it is non-perpetual and finished.
+			incentivesKeeper.SetGroup(s.Ctx, withGroupGaugeId(defaultGroup, tc.groupGauge.Id))
+
+			// Note: to avoid chances of mutations that could affect the final assertions.
+			inputCopy := deepCopyGauge(tc.groupGauge)
+
+			// Prefund the incentives module account with the coins needed for the test to succeed.
+			if tc.amountToPrefund.IsValid() {
+				s.FundModuleAcc(types.ModuleName, tc.amountToPrefund)
+			}
+
+			originalIncentivesModuleBalance := s.App.BankKeeper.GetAllBalances(s.Ctx, s.App.AccountKeeper.GetModuleAddress(types.ModuleName))
+
+			// System under test
+			err := incentivesKeeper.HandleGroupPostDistribute(s.Ctx, inputCopy, tc.coinsDistributed)
+
+			if tc.expectError {
+				s.Require().Error(err, "test: %s", name)
+				return
+			}
+			s.Require().NoError(err, "test: %s", name)
+
+			finalIncentivesModuleBalance := s.App.BankKeeper.GetAllBalances(s.Ctx, s.App.AccountKeeper.GetModuleAddress(types.ModuleName))
+
+			if !tc.groupGauge.IsPerpetual && tc.groupGauge.IsLastNonPerpetualDistribution() {
```

Comment:
Might be out of scope for this PR, but it would be good to eventually add a test covering a non-perpetual gauge that starts with having e.g. 10 epochs left and then runs 10 epochs of distributions after which we assert that it's empty and cleared from state

Reply:
Added here, good idea: [850b041](https://github.com/osmosis-labs/osmosis/pull/6509/commits/850b0415ff4ed996b1491fc58debe4245c12c766)